rust_src/src/lisp.rs

```diff
@@ -390,13 +390,32 @@ impl LispObject {
     }
 
     #[inline]
-    pub fn as_natnum_or_error(self) -> EmacsInt {
+    pub fn as_natnum(self) -> Option<EmacsUint> {
         if self.is_natnum() {
-            unsafe { self.to_fixnum_unchecked() }
+            Some(unsafe { self.as_natnum_unchecked() })
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    pub fn as_natnum_or_error(self) -> EmacsUint {
+        if let Some(num) = self.as_natnum() {
```

Comment:
Same as in the string review, how about `unwrap_or_else` like elsewhere in the code.

Reply:
I'll update it. 

As an aside, one thing I've been toying around in my head is a macro to auto generate all these as_*_or_error/as_*/as_*_unchecked APIs, since it's a lot of repetition, the the style of implementation differs across the different types. That obviously won't be part of this PR, but it's something I think is worth discussing in the future.

Reply:
:+1:

I have been chewing at uniformity. But some kind of `make_type!` sounds like a plan.